xclim/sdba/adjustment.py

```diff
@@ -228,16 +253,46 @@ def _train(self, ref, hist):
             description="Scaling factor making the mean of hist match the one of hist.",
         )
 
-    def _adjust(self, sim, degree=0):
+    def _adjust(self, sim, interp="nearest", extrapolation="constant", detrend=1):
+        # Apply preliminary scaling from obs to hist
         sim = apply_correction(
             sim,
-            broadcast(self.ds.scaling, sim, group=self.group, interp=self.interp),
+            broadcast(self.ds.scaling, sim, group=self.group, interp=interp),
             self.kind,
         )
-        sim_fit = PolyDetrend(degree=degree, kind=self.kind).fit(sim)
-        sim_detrended = sim_fit.detrend(sim)
-        scen_detrended = super()._adjust(sim_detrended)
-        scen = sim_fit.retrend(scen_detrended)
+
+        # Normalize sim group-wise
+        # This group-wise pre normalization + reapplication further down
+        # is to circumvent #442 (detrending is not groupwise)
+        mu_sim = self.group.apply("mean", sim)
+        sim_norm = apply_correction(
+            sim,
+            broadcast(
+                invert(mu_sim, kind=self.kind), sim, group=self.group, interp=interp,
+            ),
+            kind=self.kind,
+        )
+
+        # Find trend on sim (for debugging purpose, the detrending is overridable)
+        if isinstance(detrend, int):
+            detrend = PolyDetrend(degree=detrend, kind=self.kind)
+
```

Comment:
Catch case where detrend is not an int.

Reply:
Hum, but isn't that case covered by the "AttributeError" that would be raised for any objects other than BaseDetrend? I mean it you pass a string, it will raise AttributeError Object of type string has not attribute `fit` which is explicit enough for me!

Reply:
You're right.